### ISSUE TITLE
NEED HELP! Empty Batches (([], [], [], [])) During Validation

### ISSUE THREAD

**Issue by heavengu** [huggingface/nanoVLM#171]
During distributed validation, I occasionally observe that the DataLoader yields empty batches in the form of `([], [], [], [])` or dictionaries with empty values (e.g., empty tensors or lists). This causes problems in the evaluation loop, such as skipping samples or triggering runtime errors.

The validation dataset has been verified to contain no corrupted or illegal samples. I suspect this issue may be related to padding the dataset or the behavior of the `collate_fn`

Data load logic
```python
# Wrap with custom Dataset
val_dataset = VQADataset(val_ds, tokenizer, image_processor, mp_image_token_length)

# Create sampler and DataLoader
val_sampler = DistributedSampler(
    val_dataset,
    num_replicas=get_world_size(),
    rank=get_rank(),
    shuffle=False,
)
val_loader = DataLoader(
    val_dataset,
    batch_size=1,
    sampler=val_sampler,
    collate_fn=vqa_collator,
    num_workers=8,
    pin_memory=True,
    drop_last=True,
)
```
